lib/sendgrid/base_interface.rb

```diff
@@ -16,9 +16,15 @@ class BaseInterface
   #                              in the "On-Behalf-Of" header
   #   - +http_options+ -> http options that you want to be globally applied to each request
   #
-  def initialize(auth:, host:, request_headers: nil, version: nil, impersonate_subuser: nil, http_options: {})
+  def initialize(auth:, host: nil, region: nil, request_headers: nil, version: nil, impersonate_subuser: nil, http_options: {})
     @auth = auth
-    @host = host
+    if host.nil? && region.nil?
+      @host = "https://api.sendgrid.com"
+    elsif !region.nil?
+      data_residency(region: region)
+    else
+      update_host(host: host)
```

Comment:
Why is this method required? If host is provided, it'll anyways get set in line # 42.

Reply:
fixed this